src/cbmc/cbmc_parse_options.cpp

```diff
@@ -835,16 +835,21 @@ bool cbmc_parse_optionst::process_goto_program(
     // Similar removal of java nondet statements:
     // TODO Should really get this from java_bytecode_language somehow, but we
     // don't have an instance of that here.
-    const auto max_nondet_array_length=
+    const size_t max_nondet_array_length=
       cmdline.isset("java-max-input-array-length")
-        ? std::stoi(cmdline.get_value("java-max-input-array-length"))
+        ? std::stoul(cmdline.get_value("java-max-input-array-length"))
         : MAX_NONDET_ARRAY_LENGTH_DEFAULT;
+    const size_t max_nondet_tree_depth=
```

Comment:
As per these commits, gen_nondet_init stops only when **both** of the following criteria are met:

- a type is found in the `recursive_set`
- we are initializing an object accessible after a number of pointer dereferences >= `max_nondet_tree_depth`

This fact is explained in detail in the (only) place where this parameter is used in `java_object_factory.cpp`. I extended the output of `--help`.